sycl/CMakeLists.txt

```diff
@@ -16,10 +16,10 @@ if (NOT SYCL_COVERAGE_PATH)
   set(SYCL_COVERAGE_PATH "${CMAKE_CURRENT_BINARY_DIR}/profiles")
 endif()
 
-# If SYCL_ENABLE_PLUGINS is undefined, we default to enabling OpenCL and Level
-# Zero plugins.
-if (NOT DEFINED SYCL_ENABLE_PLUGINS)
```

Comment:
Should we emit a CMake warning if `SYCL_ENABLE_PLUGINS` is used? We could even set `SYCL_ENABLE_BACKENDS` to the `SYCL_ENABLE_PLUGINS` value at the same time to ease the transition.

Reply:
Yeah definitely, I did think this would maybe need to be communicated somehow so a warning is a good idea for that! Would we then just fully remove it in e.g. a months time?

Reply:
I've added this warning and value setting now.

Reply:
@kbenzie does this look okay?

Reply:
LGTM